encoding/src/adapters/mod.rs

```diff
@@ -0,0 +1,27 @@
+pub mod rle_lossless;
+pub trait PixelRWAdapter {
+  /// Decode byte stream
+  fn decode(&self, reader: &[u8]) -> Vec<u8>;
+
+  /// Write byte stream
+  fn encode(&self, writer: &[u8]) -> Vec<u8>;
```

Comment:
You are right! This interface is way too simple. I'll implement a basic example using RLE Lossless to see what works.

I think for decoding we should maybe accept a `PixelSequence`. We should also support decoding frame-by-frame. Otherwise it would be a bad experience decoding a 700MB dicom.

So maybe something like this:

```rust
#[derive(Debug, Clone, PartialEq)]
pub struct PixelSequence<'a> {
    pub offset_table: &'a C<u8>,
    pub fragments: &'a C<Vec<u8>>
}
```

I'm trying to losen the constraints as much as possible without having to copy pixel data when it is not needed.

And then decode it:

```rust
fn decode(&self, src: &PixelSequenceType, frame: u16, dst: &mut Vec<u8>) -> DecodeResult<()>;
```

For encoding I still need to come up with a solution.

Reply:
> So maybe something like this:

Passing that `PixelSequence` value to the decoding process would still require the whole raw pixel data to exist in memory. Although this is currently true for the existing DICOM object implementation (`InMemoryDicomObject`), it will not be the case for the upcoming lazy DICOM object.

Looking back to some loose ideas on this subject, I think the original intent of having this new `PixelDataObject` trait was so that a trait object could be passed to the `decode` method. Let's try this:

```rust
fn decode(&self, src: &dyn PixelDataObject, frame: u16, dst: &mut Vec<u8>) -> DecodeResult<()>;
```

On a minor side note, wouldn't `u32` be a more suitable type for the frame number?